src/components/Patient/symptoms/list.tsx

```diff
@@ -39,27 +40,46 @@ export function SymptomsList({
   const { t } = useTranslation();
 
   const [showEnteredInError, setShowEnteredInError] = useState(false);
-  const { data: symptoms, isLoading } = useQuery({
+
+  const {
+    data: symptomsData,
+    isLoading,
+    fetchNextPage,
+    hasNextPage,
+    isFetchingNextPage,
+  } = useInfiniteQuery({
     queryKey: ["symptoms", patientId, encounterId],
-    queryFn: query(symptomApi.listSymptoms, {
-      pathParams: { patientId },
-      queryParams: {
-        encounter: encounterId,
-        ordering: "-created_date",
-      },
-    }),
+    queryFn: async ({ pageParam = 0, signal }) => {
+      const response = await query(symptomApi.listSymptoms, {
+        pathParams: { patientId },
+        queryParams: {
+          encounter: encounterId,
+          ordering: "-created_date",
+          limit: showTimeline ? 30 : 14,
+          offset: String(pageParam),
+        },
+      })({ signal });
+      return response as PaginatedResponse<Symptom>;
+    },
+    initialPageParam: 0,
+    getNextPageParam: (lastPage, allPages) => {
+      const currentOffset = allPages.length * 14;
+      return currentOffset < lastPage.count ? currentOffset : null;
+    },
   });
 
   if (isLoading) {
     return <TableSkeleton count={5} />;
   }
 
-  const filteredSymptoms = symptoms?.results?.filter(
+  const symptoms = symptomsData?.pages.flatMap((page) => page.results) ?? [];
+
+  const filteredSymptoms = symptoms?.filter(
```

Comment:
Same here, can do the same thing as diagnosis since filters are condition spec based.